packages/renderer/src/lib/objects/KubernetesObjectsList.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import type { KubernetesObject } from '@kubernetes/client-node';
+import type { TableColumn, TableRow } from '@podman-desktop/ui-svelte';
+import { Button, FilteredEmptyScreen, NavPage, Table } from '@podman-desktop/ui-svelte';
+import { onDestroy, onMount, type Snippet } from 'svelte';
+import { type Readable, type Unsubscriber, type Writable } from 'svelte/store';
+
+import type { IDisposable } from '../../../../main/src/plugin/types/disposable';
+import { withBulkConfirmation } from '../actions/BulkActions';
+import KubeActions from '../kube/KubeActions.svelte';
+import { listenResources } from '../kube/resources-listen';
+import KubernetesCurrentContextConnectionBadge from '../ui/KubernetesCurrentContextConnectionBadge.svelte';
+
+export interface Kind {
+  resource: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  transformer: (o: KubernetesObject) => any;
+  delete: (name: string) => Promise<void>;
+  isResource: (o: KubernetesObject) => boolean;
+  legacySearchPatternStore: Writable<string>;
+  legacyObjectStore: Readable<KubernetesObject[]>;
+}
+
+interface Props {
+  kinds: Kind[];
+  searchTerm: string;
+  singular: string;
+  plural: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  columns: TableColumn<any>[];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  row: TableRow<any>;
+
+  emptySnippet: Snippet;
+}
+
+let { kinds, singular, plural, icon, searchTerm, columns, row, emptySnippet }: Props = $props();
+
+let resources = $state<{ [key: string]: KubernetesObject[] | undefined }>({});
+let resourceListeners: (IDisposable | undefined)[] = [];
+let legacyUnsubscribers: Unsubscriber[] = [];
+
+const objects = $derived(
+  kinds.flatMap(kind => resources[kind.resource]?.map(object => kind.transformer(object)) ?? []),
+);
+
+$effect(() => {
+  kinds.forEach(kind => kind.legacySearchPatternStore.set(searchTerm));
+});
+
+onMount(async () => {
+  for (let kind of kinds) {
+    resourceListeners.push(
+      await listenResources(
+        kind.resource,
+        {
+          searchTermStore: kind.legacySearchPatternStore,
+        },
+        (updatedResources: KubernetesObject[]) => {
+          resources[kind.resource] = updatedResources;
+        },
+      ),
+    );
+
+    legacyUnsubscribers.push(kind.legacyObjectStore.subscribe(o => (resources[kind.resource] = o)));
```

Comment:
Received an error repeatedly while testing:
`Assignment to `secrets` property (packages/​​renderer/​​src/​​lib/​​objects/​​KubernetesObjectsList.svelte:68:68) will evaluate to the right-hand side, not the value of `secrets` following the assignment. This may result in unexpected behaviour.https://svelte.dev/e/assignment_value_stale`

I think you meant to have { } brackets instead of ( ): `{ resources[kind.resource] = o })`.

Reply:
Thanks for the catch. Fixed

Reply:
> Thinking about it a bit more I don't think https://github.com/podman-desktop/podman-desktop/issues/10916 would directly help, because the objects passed to delete are already transformed (like PVCUI, PodUI, etc) and not the underlying Kube object. We could also just keep track internally of which resource array each object came from.

> Either way I think we can remove the need for isResource, but it is not blocking. I'm fine if you can open a follow-up issue.

-> #11002